cpp/src/binaryop/compiled/binary_ops.cu

```diff
@@ -375,6 +385,79 @@ void binary_operation(mutable_column_view& out,
   auto [rhsv, aux] = scalar_to_column_view(rhs, stream);
   operator_dispatcher(out, lhs, rhsv, false, true, op, stream);
 }
+
+namespace detail {
+void apply_sorting_struct_binary_op(mutable_column_view& out,
+                                    column_view const& lhs,
+                                    column_view const& rhs,
+                                    bool is_lhs_scalar,
+                                    bool is_rhs_scalar,
+                                    binary_operator op,
+                                    rmm::cuda_stream_view stream)
+{
+  CUDF_EXPECTS(lhs.type().id() == type_id::STRUCT && rhs.type().id() == type_id::STRUCT,
+               "Both columns must be struct columns");
+  CUDF_EXPECTS(!cudf::structs::detail::is_or_has_nested_lists(lhs) and
+                 !cudf::structs::detail::is_or_has_nested_lists(rhs),
+               "Lists not supported");
+  // Struct child column type and structure mismatches are caught within the two_table_comparator
+  switch (op) {
```

Comment:
The side effect of this change is you lose the default constructor for the comparator argument and push down some of the type checking down into the base detail functions. I don't think this is a deal breaker because we're already in detail namespace. Should something similar be done to `operator_dispatcher` in the same file?

Reply:
Actually, the bigger issue here is that it templates the function and necessitates it move to a header file. Need to figure out how this would work with the spark side since that last commit broke some stuff...

Reply:
Aha. You’re right, my suggestion is a little awkward. Losing the default argument was something I considered but didn’t see as an issue because we were already supplying that value explicitly (because of the stream that needs to be passed through). However, your notes about type checking and moving it to a header are enough reason to ignore my proposal and keep this as-is.

Reply:
I've reverted the changes. If we want to do this in the future it would require moving the dispatchers to a `.cuh` and then creating a wrapper declared in `ColumnViewJni.hpp` and declared in `ColumnViewJni.cu` that is used by `ColumnViewJni.cpp`.